### PR TITLE
Fix FreeBSD build failures. Update tests for FreeBSD

### PR DESCRIPTION
**Pull request by Thefrank**
# Fix FreeBSD build failures. Update tests for FreeBSD

## Changes
- src/libpsl-native/CMakeLists.txt -> add CMake case for FreeBSD
- src/libpsl-native/src/getcurrentthreadid.cpp -> add FreeBSD case
- src/libpsl-native/src/getppid.cpp -> rework function, add case for FreeBSD 
- src/libpsl-native/src/getuserfrompid.cpp -> fix FreeBSD build failure from missing header
- src/libpsl-native/test/ -> add FreeBSD to Apple cases. Apple and FreeBSD use the same parameters for `stat`

There are still two test failures related to GetDeviceID. FreeBSD returns a nonsensically high number for `/usr/bin/stat -f %d /`

Ubuntu 20.04:
```
frank@buildbot:~$ /usr/bin/stat -c %d /
64768
```

FreeBSD 13.1:
```
root@freebsd:~ # /usr/bin/stat -f %d /
2870232080986416670
```

### Reasoning
Current FreeBSD builds fail. This PR resolves it and adds supporting tests

### CODE REVIEW

**Comment by Thefrank** [comment]
@microsoft-github-policy-service agree

**Comment by Thefrank** [comment]
Not sure who I should get code review from / who has write access so I am guessing based on recent PR merges

PTAL @adityapatwardhan

**Comment by Thefrank** [comment]
If there is nothing else, then this is ready to merge

**Comment by shanselman** [comment]
@SteveL-MSFT this is the last one I think that's needed for https://github.com/dotnet/runtime/issues/14537 upstream

File: src/libpsl-native/CMakeLists.txt
Comment (emaste) [inline src/libpsl-native/CMakeLists.txt:13]: It seems strange to pass `-Wl` with no arg. Of course the oddity was just duplicated from the Darwin case (where it exists in the first public version of this file).

It seems to me we ought to just mimic the Linux behaviour here, e.g. make the existing case above `if (${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR ${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")`

File: src/libpsl-native/src/getcurrentthreadid.cpp
Comment (emaste) [inline src/libpsl-native/src/getcurrentthreadid.cpp:26]: This cast is superfluous - this function returns `pid_t`, which is the type of `tid`. Probably want in the line above `tid = (pid_t)pthread_getthreadid_np();`, or just avoid explicit casts altogether.

File: src/libpsl-native/src/getppid.cpp
Comment (emaste) [inline src/libpsl-native/src/getppid.cpp:34]: roll this commit into previous?

File: src/libpsl-native/CMakeLists.txt
Comment (Thefrank) [reply src/libpsl-native/CMakeLists.txt:13]: I found it unusual too, but opted to include it as I assumed if it was important enough for `"Darwin"` to have it then so should `"FreeBSD"`. I will go ahead and add `"FreeBSD"` to the `"Linux"` case as suggested. Thanks!

File: src/libpsl-native/src/getcurrentthreadid.cpp
Comment (Thefrank) [reply src/libpsl-native/src/getcurrentthreadid.cpp:26]: This one...I tried my best here to follow what is done under dotNET for `pthread_getthreadid_np()`  but found it [inconsistent](https://github.com/search?q=repo%3Adotnet%2Fruntime%20pthread_getthreadid_np&type=code). I can make changes both to this PR and open PRs in runtime to correct the usage it in order to avoid any information loss due to casting.

**Review by emaste — APPROVED** [review]

**Comment by Thefrank** [comment]
What else is needed to get this merged?

Any feedback would be great!

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@Thefrank  thank you for your contribution!

**Pull request merged**
